### PR TITLE
WIP: hybrid tests

### PR DESCRIPTION
**Pull request by krlmlr**
Currently failing.

### CODE REVIEW

File: tests/testthat/test-hybrid-traverse.R [hunk 1]
```diff
@@ -9,24 +9,136 @@ test_df <- data_frame(
...
+  expect_equal(
+    test_df %>%
+      grouping %>%
+      mutate(f = { xx <- 5; xx }) %>%
+      select(-e),
+    test_df %>%
+      mutate(f = 5) %>%
+      grouping %>%
+      select(-e))
+})
```
Comment (hadley) [inline tests/testthat/test-hybrid-traverse.R]: I think we should add this just to be explicit:

```R
expect_false(exists("xx")
```

**Comment by hadley** [comment]
I'm not sure if this is important or not:

```R
df <- tibble(
  x = c("a", "b")
  y = list(list(a = 1, b = 2), list(a = 1, b = 2)),
)

df %>% rowwise() %>% mutate(z = y[[x]])
```

I can't quite see when this would be useful except in conjunction with `rowwise()`, which is going away.

**Comment by hadley** [comment]
I'm happy to break in favour of the lazyeval API. They were never obviously documented, so I don't expect many people will be using them.

**Comment by krlmlr** [comment]
Closing in favor of #2190.

**Pull request closed without merging**